packages/serve/src/models/extensions/authenticator.ts

```diff
@@ -0,0 +1,52 @@
+import { VulcanExtension, ExtensionBase } from '@vulcan-sql/core';
+import { KoaContext } from '@vulcan-sql/serve/models';
+import { TYPES } from '../../containers/types';
+
+export interface AuthUserInfo {
+  name: string;
+  attr: {
+    [field: string]: string | boolean | number;
+  };
+}
+
+export enum AuthStatus {
+  /**
+   * SUCCESS: Request format correct and match the one of user credentials
+   * INCORRECT: Request format is incorrect for authenticator needed, skip and check next authenticator
+   * FAIL: Request format correct, but not match the user credentials
+   */
+  SUCCESS = 'SUCCESS',
+  FAIL = 'FAIL',
+  INCORRECT = 'INCORRECT',
```

Comment:
NIT: Can we use other words like "indeterminate", "skipped", "bypass" instead of "incorrect"? "Incorrect" makes me think about "bad credentials", and "fail" means failure to authenticate due to internal issues.

Reply:
Thanks for the suggestion! I chose the `indeterminate` for renaming,